### PR TITLE
Pass raw data with ContractLogicError

### LINKED ISSUE #2793: Add support for custom errors or provide full RPC response if transaction is reverted
* Version: 6.0.0b9
* Python: 3.10.9
* OS: osx

### What was wrong?
I need to call a contract method and get revert data
```
from web3 import Web3
w3 = Web3(Web3.HTTPProvider("https://eth-goerli.public.blastapi.io"))
contract = w3.eth.contract(address=address, abi=abi)
contract.functions.alwaysReverts().call()
```
**Unfortunately** the exception raised with `call()` method provides no information about triggered custom error and the data passed as the error parameters:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  ...
    File "/Users/alexander/Desktop/FastAPI/venv/lib/python3.10/site-packages/web3/_utils/method_formatters.py", line 636, in     raise_solidity_error_on_revert
    raise ContractLogicError(response["error"]["message"])
web3.exceptions.ContractLogicError: execution reverted
```

If we look at the variable `response` used to raise the exception, we see that this data is contained in `response["error"]["data"]` 
```
>>> response
{'jsonrpc': '2.0', 'id': 1, 'error': {'code': 3, 'message': 'execution reverted', 'data': '0x8f046edb0000000000000000000000000000000000000000000000000000000063d1a1a400000000000000000000000000000000000000000000000000000000000003ff'}}
```

But it isn't used in the exception and there is no access to raw response data but there should be a proper way to get it

### PR DESCRIPTION
**Pull request by kclowes**
### What was wrong?

People frequently request the raw data to be sent back with the ContractLogicError. 

Closes #2793, #2336 

### How was it fixed?

This PR sends back the raw data as an attribute on the error class without breaking the existing API. 

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)
- [x] Documentation

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Review by fselmo — APPROVED** [review]
Oh I like this approach in general! The more information the better on exceptions 👌🏼. 

I think we should be able to add this to the ``OffchainLookup`` exception also which would be some nice information to have available.

File: tests/integration/test_ethereum_tester.py [hunk 1]
```diff
@@ -596,6 +609,36 @@ def test_eth_get_balance_with_block_identifier(self, w3: "Web3") -> None:
...
+        self,
+        w3: "Web3",
+        offchain_lookup_contract: "Contract",
+    ) -> None:
+        # test ContractFunction call
+        with pytest.raises(OffchainLookup) as e:
+            offchain_lookup_contract.functions.testOffchainLookup(
+                OFFCHAIN_LOOKUP_TEST_DATA
+            ).call(ccip_read_enabled=False)
+        assert e.value.data == OFFCHAIN_LOOKUP_RETURN_DATA
```
Comment (kclowes) [inline tests/integration/test_ethereum_tester.py:622]: @fselmo this assertion fails if it's run in isolation. the `OFFCHAIN_LOOKUP_RETURN_DATA` changes, but only fails that way on eth-tester, not with the geth integration tests. Do you have any ideas why the `OFFCHAIN_LOOKUP_RETURN_DATA` would be different when the whole file is run vs. run in isolation?

Comment (fselmo) [reply tests/integration/test_ethereum_tester.py:622]: Oh, yeah. The very first value that's returned is the [particular contract's address](https://github.com/ethereum/web3.py/blob/c8ef104fd428972567ed9475eb8ac7b28e4be261/web3/_utils/contract_sources/OffchainLookup.sol#L18) which will differ between the contract deployed via the geth fixture and the one for eth-tester. You'll just have to change the first 20 bytes of the `OFFCHAIN_LOOKUP_RETURN_DATA` to match that.

Comment (kclowes) [reply tests/integration/test_ethereum_tester.py:622]: Aha, thank you! I'll update the assertion so that it only asserts on `e.value.data[:74]`

Comment (fselmo) [reply tests/integration/test_ethereum_tester.py:622]: Or just update [10:74] to match the address that deploys to the `eth-tester` fixture

Comment (fselmo) [reply tests/integration/test_ethereum_tester.py:622]: Changing `OFFCHAIN_LOOKUP_RETURN_DATA`... should be:

```py
OFFCHAIN_LOOKUP_RETURN_DATA = "0x556f1830000000000000000000000000f2e246bb76df876cef8b38ae84130f4f55de395b00000000000000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000000001c0da96d05a0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002400000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000000000000000000000000000000000000000002c68747470733a2f2f776562332e70792f676174657761792f7b73656e6465727d2f7b646174617d2e6a736f6e0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002568747470733a2f2f776562332e70792f676174657761792f7b73656e6465727d2e6a736f6e00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000001474657374206f6666636861696e206c6f6f6b757000000000000000000000000000000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000001474657374206f6666636861696e206c6f6f6b7570000000000000000000000000"
```

**Review by pacrob — APPROVED** [review]
lgtm!

**Pull request merged**